packages/cozy-stack-client/src/DocumentCollection.js

```diff
@@ -533,7 +533,7 @@ The returned documents are paginated by the stack.
     // We need to pass the partialFilter in the selector, otherwise CouchDB might
     // fallback on another index if it does not exist yet
     const mergedSelector = partialFilter
-      ? merge(selector, partialFilter)
+      ? merge({ ...selector }, partialFilter)
```

Comment:
can we add a test for that? Having a test will help us... next time to remove the need of lodash without fearing any regression on this kind of stuff

Reply:
Fair enough, I'll add it

Reply:
16ddffb6e3751bd150e0df93cf992e5a2ac0315d
Note the merge itself was already tested